src/main/java/org/galatea/starter/service/SettlementService.java

```diff
@@ -55,7 +55,37 @@
     return idSet;
   }
 
+
+  /**
+   * Retrieve a previously-generated settlement mission from the database.
+   *
+   * @param id the ID of the mission to retrieve
+   */
   public Optional<SettlementMission> findMission(final Long id) {
+    log.info("Retrieving settlement mission with id {}", id);
     return Optional.ofNullable(missionrpsy.findOne(id));
```

Comment:
Should this throw an EntityNotFoundException like findMissions if the mission isn't found?

Reply:
The exception is thrown `SettlementRestController#getMission` if the optional is empty.

Reply:
Created an issue to decide where the exception should be thrown: https://github.com/GalateaRaj/fuse-starter-java/issues/268